src/tests/src/solver/optimisation/adequacy_patch.cpp

```diff
@@ -32,22 +33,25 @@ std::pair<double, double> setNTCboundsForOneTimeStep(AdequacyPatchMode originTyp
 
     problem.adequacyPatchRuntimeData->originAreaMode[0] = originType;
     problem.adequacyPatchRuntimeData->extremityAreaMode[0] = extremityType;
-    problem.adqPatchParams
-      = std::unique_ptr<AdequacyPatchParameters>(new AdequacyPatchParameters());
-    auto& adqPatchParams = problem.adqPatchParams;
+    problem.adequacyPatchRuntimeData->AdequacyFirstStep = true;
 
-    adqPatchParams->AdequacyFirstStep = true;
-    adqPatchParams->SetNTCOutsideToOutsideToZero = SetNTCOutsideToOutsideToZero;
-    adqPatchParams->SetNTCOutsideToInsideToZero = SetNTCOutsideToInsideToZero;
+    AdqPatchParams adqPatchParams;
+    adqPatchParams.localMatching.setToZeroOutsideOutsideLinks = SetNTCOutsideToOutsideToZero;
+    adqPatchParams.localMatching.setToZeroOutsideInsideLinks = SetNTCOutsideToInsideToZero;
 
     VALEURS_DE_NTC_ET_RESISTANCES ValeursDeNTC;
-    ValeursDeNTC.ValeurDeNTCOrigineVersExtremite = &origineExtremite;
-    ValeursDeNTC.ValeurDeNTCExtremiteVersOrigine = &extremiteOrigine;
+    ValeursDeNTC.ValeurDeNTCOrigineVersExtremite = new double[1];
+    ValeursDeNTC.ValeurDeNTCExtremiteVersOrigine = new double[1];
+    ValeursDeNTC.ValeurDeNTCOrigineVersExtremite[0] = origineExtremite;
+    ValeursDeNTC.ValeurDeNTCExtremiteVersOrigine[0] = extremiteOrigine;
 
-    double Xmin;
-    double Xmax;
+    double Xmin(0.);
+    double Xmax(0.);
 
-    setNTCbounds(Xmax, Xmin, &ValeursDeNTC, 0, &problem);
+    setNTCbounds(Xmax, Xmin, &ValeursDeNTC, 0, &problem, adqPatchParams);
```

Comment:
It's quite clear that `setNTCbounds` only needs a `LocalMatching` object. Can you change it's definition to take this into account ?

Reply:
Here again, we use `adqPatchParams.enabled`, so we can't really pass `adqPatchParams.localMatching`